### PR TITLE
Factor out BLSCT wallet code

### PR DESCRIPTION
**Pull request by aguycalled**

### CODE REVIEW

**Comment by mxaddict** [comment]
Took a quick look at the PR, seems good, will wait for the builds to finish

File: src/blsct/wallet/keyman.cpp [hunk 1]
```diff
@@ -195,13 +194,12 @@ void KeyMan::SetHDSeed(const PrivateKey& key)
...
-    auto blindingKey = PrivateKey(BLS12_381_KeyGen::derive_child_SK(childKey, 1));
-    auto tokenKey = PrivateKey(BLS12_381_KeyGen::derive_child_SK(childKey, 2));
-    auto viewKey = PrivateKey(BLS12_381_KeyGen::derive_child_SK(transactionKey, 0));
-    auto spendKey = PrivateKey(BLS12_381_KeyGen::derive_child_SK(transactionKey, 1));
+    auto childKey = FromSeedToChildKey(key.GetScalar());
+    auto transactionKey = FromChildToTransactionKey(childKey);
+    auto blindingKey = PrivateKey(FromChildToBlindingKey(childKey));
+    auto tokenKey = PrivateKey(FromChildToTokenKey(childKey));
+    auto viewKey = PrivateKey(FromTransactionToViewKey(transactionKey));
+    auto spendKey = PrivateKey(FromTransactionToSpendKey(transactionKey));
```
Comment (exfinen) [inline src/blsct/wallet/keyman.cpp:202]: Perhaps, it's more consistent to call this = `spendingKey` as called in `TxOutBLSCTData`

**Review by exfinen — APPROVED** [review]

File: src/blsct/wallet/keyman.cpp (hunk 1 above)
Comment (aguycalled) [reply src/blsct/wallet/keyman.cpp:202]: its a different key, not the same from the output

Comment (exfinen) [reply src/blsct/wallet/keyman.cpp:202]: I see.

**Comment by mxaddict** [comment]
The failing tests are related to branding issue same as on master branch, I think this PR is good to merge.

**Review by mxaddict — APPROVED** [review]

**Pull request merged**